src/Plugin/Plugins/TempHumidPlugin.h

```diff
@@ -0,0 +1,242 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+/*******************************************************************************
+    DESCRIPTION
+*******************************************************************************/
+/**
+ * @brief  Reads temperature and humidity from a sensor and displays it
+ * @author Flavio Curti <fcu-github@no-way.org>
+ * @addtogroup plugin
+ *
+ * @{
+ */
+
+#ifndef __TEMPHUMIDPLUGIN_H__
+#define __TEMPHUMIDPLUGIN_H__
+
+/******************************************************************************
+ * Compile Switches
+ *****************************************************************************/
+#define DHTTYPE DHTesp::DHT11   // DHT 11 sensor
+#define DHTTYPE DHTesp::DHT11   // DHT 11 sensor
+
+
+/******************************************************************************
+ * Includes
+ *****************************************************************************/
+#include <stdint.h>
+#include "Plugin.hpp"
+
+#include <DHTesp.h>
+#include <SimpleTimer.hpp>
+
+#include <Canvas.h>
+#include <BitmapWidget.h>
+#include <TextWidget.h>
+
+/******************************************************************************
+ * Macros
+ *****************************************************************************/
+
+/******************************************************************************
+ * Types and Classes
+ *****************************************************************************/
+
+/**
+ * TempHumid plugin.
+ *
+ * This plugins displays the temperature and humidity read from a sensor
+ * currently implemented sensors:
+ * - DHT Sensors supported by DHTesp library
+ * - TODO: sht3x
+ */
+class TempHumidPlugin : public Plugin
+{
+public:
+
+    /**
+     * Constructs the plugin.
+     *
+     * @param[in] name  Plugin name
+     * @param[in] uid   Unique id
+     */
+    TempHumidPlugin(const String& name, uint16_t uid) :
+        Plugin(name, uid),
+        m_textCanvas(nullptr),
+        m_iconCanvas(nullptr),
+        m_bitmapWidget(),
+        m_textWidget("\\calign?"),
+        m_text(),
+        m_page(0U),
+        m_pageTime(),
+        m_timer(),
+        m_xMutex(nullptr),
+        m_dht(),
+        m_humid(0),
+        m_temp(0),
+        m_last(0),
```

Comment:
Please assign type safe values, e.g. for float please use "0.0F" and "0U" for unsigned long.